### ISSUE TITLE
DeviceClient never confirms message delivery

### ISSUE THREAD

**Issue by aromaa** [nanoframework/Home#935]
### Details about Problem

**.NET nanoFramework library:** nanoFramework.Azure.Devices / nanoFramework.m2mqtt

### Description
DeviceClient never confirms delivery of SendMessage or UpdateReportedProperties and waits for the cancellation token cancellation.

**Comment by Ellerbach** [comment]
Can you please give the code you are using?
If you are using AtMostOnce (0), it's normal you'll never ever receive any confirmation as in this mode, you'll not get any!
So you need level 1 to get a confirmation.

**Comment by aromaa** [comment]
Added the repro code, but yes, this doesn't work even when using AtLeastOnce. And UpdateReportedProperties uses always AtLeastOnce so the qos level only applies to SendMessage.

**Comment by Ellerbach** [comment]
> ```cs
> AtLeastOnce
> ```

Let me try again in next days. The only issue I saw in the past was when the network is too fast. You get the confirmation before it is added in the queue. All is using the same principal. So that *maybe* your case.

**Comment by Ellerbach** [comment]
> Added the repro code, but yes, this doesn't work even when using AtLeastOnce. And UpdateReportedProperties uses always AtLeastOnce so the qos level only applies to SendMessage.

Also, just to be 100% sure, you're using the latest preview version?

**Comment by aromaa** [comment]
Yes, using the latest preview version. The network conditions are indeed good and suspected it too when I took a quick glance on the code.

**Comment by Ellerbach** [comment]
> Yes, using the latest preview version. The network conditions are indeed good and suspected it too when I took a quick glance on the code.

So seems it's the network which is too fast :-). I will look at how to kind of force to get it and adjust the code to add it before. Will play with that in next few days.

**Comment by Ellerbach** [comment]
@aromaa can you please check if the PR https://github.com/nanoframework/nanoFramework.Azure.Devices/pull/65 fixes your issue? Clone the repo, checkout the branch and rather than using the nuget, if you can use that code, would be great.
I'll test it as well later.

**Comment by aromaa** [comment]
Seems not to be enough, not getting any confirmations though

**Comment by Ellerbach** [comment]
> Seems not to be enough, not getting any confirmations though

Good news: I manage to reproduce the issue. Bad news: that's insane to debug!

**Comment by Ellerbach** [comment]
@aromaa feel free to test the new version, all should now work fine. I've tested your example, works like a charm for me again in all circumstances. Thanks for finding out and thanks for the sample code!

**Comment by aromaa** [comment]
Can confirm that the issue has been fixed! Yet another thing off backlog :)
